e3/vcs/svn.py

```diff
@@ -0,0 +1,166 @@
+
+        Add the non-interactive option to all command (accepted on all SVN.
+        subcommands from version 1.5).
+        :param cmd: the command line as a list of string, all None entries will
+            be discarded
+        :type cmd: list[str | None]
+        :param kwargs: additional parameters to provide to e3.os.process.Run
+        :type kwargs: dict
+        :return: Result of the Run of the SVN command
+        :rtype: Run object
+        :raise: SVNError
+        """
+        if self.__class__.svn_bin is None:
+            svn_binary = e3.os.process.which('svn', default=None)
+            if svn_binary is None:  # defensive code
+                raise SVNError('cannot find svn', 'svn_cmd')
+            self.__class__.svn_bin = e3.os.fs.unixpath(svn_binary)
+
+        if 'output' not in kwargs:
+            kwargs['output'] = self.log_stream
+
+        p_cmd = [arg for arg in cmd if arg is not None]
+        p_cmd.insert(0, self.__class__.svn_bin)
+        p_cmd.append('--non-interactive')
+
+        p = e3.os.process.Run(p_cmd, cwd=self.working_copy, **kwargs)
+        if p.status != 0:
+            raise SVNError('%s failed (exit status: %d)' %
+                           (e3.os.process.command_line_image(p_cmd), p.status),
+                           origin='svn_cmd', process=p)
+        return p
+
+    def get_info(self, item):
+        """Return a specific item shown by svn info.
+
+        The --show-item option is only available from 1.9.
+        :rtype: str
+        :raise: SVNError
+        """
+        info = self.svn_cmd(['info'], output=PIPE).out
+        m = re.search(r'^{item}: *(.*)\n'.format(item=item), info, flags=re.M)
+        return m.group(1).strip()
+
+    @property
+    def url(self):
+        """Return the last URL used for the checkout.
+
+        :rtype: str
+        :raise: SVNError
+        """
+        return self.get_info('URL')
+
+    @property
+    def current_revision(self):
+        """Return the current revision.
+
+        :rtype: str
+        :raise: SVNError
+        """
+        return self.get_info('Revision')
+
+    def update(self, url=None, revision=None, force_and_clean=False):
+        """Update a working copy or checkout a new one.
+
+        If the directory is already a checkout, it tries to update it.
+        If the directory is not associated to a (good) checkout or is empty
+        it will checkout.
+        The option --remove-unversioned of the svn subcommand
+        cleanup exists only from svn version 1.9.
+        :param url: URL of a SVN repository
+        :type url: str
+        :param revision: specific revision (default is last)
+        :type revision: str | None
+        :param force_and_clean: if True: erase the content of non empty
+        working_copy and use '--force' option for the svn update/checkout
+        command
+        :type force_and_clean: bool
+        :raise: SVNError
+        """
+        def is_clean_svn_dir(dir_path, ignore_change=True):
```

Comment:
Having the ignore_change in that function makes the logic a little too complex IMHO. I would prefer to remove it here and always return True, False when a svn dir is not clean. Then you can perfectly ignore the "not clean" status when not in force_and_clean.